### PR TITLE
adding settings file lookup for default oauth keys

### PR DESCRIPTION
**Pull request by flyinbutrs**
Adding local OAuth creds detection as in hostgroup provider PR #366

### CODE REVIEW

File: lib/puppet/provider/foreman_smartproxy/rest_v2.rb [hunk 1]
```diff
@@ -13,12 +13,32 @@ def self.specificity
   end
 
   def api
+    if resource[:consumer_key]
+      key = resource[:consumer_key]
+    else
+      begin
+        key = YAML.load_file('/etc/foreman/settings.yaml')[:oauth_consumer_key]
+      rescue
+        fail "Hostgroup #{resource[:name]} cannot be registered: No OAUTH Consumer Key available"
```
Comment (mmoll) [inline lib/puppet/provider/foreman_smartproxy/rest_v2.rb]: it's not a Hostgroup here ;)

File: lib/puppet/provider/foreman_smartproxy/rest_v2.rb [hunk 2]
```diff
@@ -13,12 +13,32 @@ def self.specificity
...
+      end
+    end
+
+    if resource[:consumer_secret]
+      secret = resource[:consumer_secret]
+    else
+      begin
+        secret = YAML.load_file('/etc/foreman/settings.yaml')[:oauth_consumer_secret]
+      rescue
+        fail "Hostgroup #{resource[:name]} cannot be registered: No OAUTH Consumer Secret available"
```
Comment (mmoll) [inline lib/puppet/provider/foreman_smartproxy/rest_v2.rb]: and here

**Comment by mmoll** [comment]
:+1: otherwise

**Comment by flyinbutrs** [comment]
D'oh! Fixed...

**Comment by mmoll** [comment]
Hm, I don't see a change in the diff?

**Comment by flyinbutrs** [comment]
Whoops...  turns out you have to add files to a commit before amending if you want the changes to show up :)

**Comment by mmoll** [comment]
:+1:

**Comment by ekohl** [comment]
Thanks!

**Pull request closed without merging**